### PR TITLE
feat(store): add zero-dependency local fallback stores and make native deps optional

### PR DESCRIPTION
**Pull request by amondnet**
## Summary

- Add `LocalVectorStore`: a JSON file-based vector store with cosine similarity search and zero native dependencies, replacing `NullVectorStore` as the default in `DefaultContextStore`
- Move `@lancedb/lancedb` to `optionalDependencies` in both `@pleaseai/rpg-store` and `@pleaseai/rpg-utils`, making LanceDB an opt-in dependency
- Remove `NullVectorStore` (`packages/store/src/null-vector-store.ts`) and the `vectorEnabled` field from `ContextStoreConfig`

## Changes

- `packages/store/src/local/vector-store.ts` — new `LocalVectorStore` implementation (JSON persistence, cosine similarity)
- `packages/store/src/local/index.ts` — barrel export for `./local` sub-path
- `packages/store/src/default-context-store.ts` — switch default vector store from LanceDB to `LocalVectorStore`
- `packages/store/src/types.ts` — remove `vectorEnabled` field from `ContextStoreConfig`
- `packages/store/package.json` — `@lancedb/lancedb` moved to `optionalDependencies`; added `./null-vector-store` and `./local` sub-path exports
- `packages/utils/package.json` — `@lancedb/lancedb` moved to `optionalDependencies`
- `packages/graph/src/rpg.ts` — remove `vectorEnabled: false` from `store.open()` call
- `tests/store/local-vector-store.test.ts` — 10 unit tests for `LocalVectorStore`

## Test Plan

- [ ] Run `bun run test tests/store/local-vector-store.test.ts` — all 10 unit tests pass
- [ ] Run `bun run build` — build succeeds without LanceDB native bindings required
- [ ] Run `bun run typecheck` — no type errors
- [ ] Verify `DefaultContextStore` works without `@lancedb/lancedb` installed

---
## Summary by cubic
Adds LocalVectorStore and local graph/text stores so the app runs without native deps. DefaultContextStore uses the local vector store and falls back to local graph/text when SQLite isn’t available; SemanticSearch injects a VectorStore and LanceDB is optional.

- **New Features**
  - LocalVectorStore with JSON persistence at {path}/vectors.json and cosine search; supports a 'memory' path, cleans up its temp dir on close, and throws on embedding dimension mismatch.
  - Added LocalGraphStore (JSON-backed) and LocalTextSearchStore (in-memory). DefaultContextStore falls back to these when better-sqlite3 isn’t available.
  - SemanticSearch now takes a VectorStore via constructor; MCP server and tests updated to use LocalVectorStore. Hybrid/FTS paths log and fall back to vector-only.
  - rpg-store: added ./local export; moved @lancedb/lancedb and better-sqlite3 to optionalDependencies. Removed the LanceDB-based VectorStore from rpg-utils.
  - Tests: 10 unit tests for LocalVectorStore plus 38 tests for LocalGraphStore/TextSearchStore; removed obsolete hybrid test.
  - Robustness: validate LocalVectorStore.open() config.path; error on corrupt vectors.json; optional clear() guarded; DefaultContextStore.close() aggregates errors; MCP checks {path}/vectors.json before reindexing; LocalGraphStore.close() cleans up its temp dir; DefaultContextStore falls back to local stores only on MODULE_NOT_FOUND errors (other SQLite errors are rethrown).

- **Migration**
  - Update SemanticSearch usage to pass a VectorStore instance; remove dbPath/tableName.
  - Remove vectorEnabled from ContextStoreConfig.
  - LanceDB and better-sqlite3 are opt-in; install them only if you prefer their engines over the local stores.

<sup>Written for commit 1f2088101524e59487ff77940b3d1150c1631ec1. Summary will update on new commits.</sup>

### CODE REVIEW

File: packages/store/src/local/vector-store.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  async close(): Promise<void> {
+    this.flush()
+    this.index = new Map()
+    this.filePath = null
+  }
+
+  async upsert(id: string, embedding: number[], metadata?: Record<string, unknown>): Promise<void> {
+    this.index.set(id, { embedding, metadata })
+    this.flush()
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/store/src/local/vector-store.ts]: ⏭️ Not applied
- Reason: The `upsertBatch` method already batches multiple writes into a single flush, covering the main hot path. Debouncing would add complexity and complicate test determinism for a zero-dependency dev-fallback store. The per-operation flush ensures durability without surprises.
- Alternative: Use `upsertBatch` for bulk operations to amortize the cost.

Thank you for the feedback, @gemini-code-assist!

File: packages/store/src/local/vector-store.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  for (let i = 0; i < len; i++) {
+    const ai = a[i] ?? 0
+    const bi = b[i] ?? 0
+    dot += ai * bi
+    normA += ai * ai
+    normB += bi * bi
+  }
+  const denom = Math.sqrt(normA) * Math.sqrt(normB)
+  return denom === 0 ? 0 : dot / denom
+}
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/store/src/local/vector-store.ts]: ✅ Applied in [04fe187](https://github.com/pleaseai/RPG/commit/04fe187)
- File: `packages/store/src/local/vector-store.ts`
- Change: Replaced `Math.min` dimension truncation with a strict equality check that throws `Error: Vector dimension mismatch: X vs Y` — matches your suggestion exactly (with `!` non-null assertions to satisfy `noUncheckedIndexedAccess`)

Thanks for the suggestion, @gemini-code-assist!
